src/main/java/model/piece/Pho.java

```diff
@@ -0,0 +1,91 @@
+import java.util.Map.Entry;
+import model.Path;
+import model.Point;
+import model.Team;
+
+public class Pho extends Piece {
+
+    public Pho(Team team) {
+        super(team);
+        pieceName = PieceName.PHO;
+    }
+
+    @Override
+    public boolean isValidPoint(Point beforePoint, Point targetPoint) {
+        boolean isStraightMove = beforePoint.x() == targetPoint.x() || beforePoint.y() == targetPoint.y();
+        boolean isSamePoint = beforePoint.x() == targetPoint.x() && beforePoint.y() == targetPoint.y();
+
+        return isStraightMove && !isSamePoint;
+    }
+
+    @Override
+    public Path calculatePath(Point beforePoint, Point targetPoint) {
+        int vectorX = targetPoint.x() - beforePoint.x();
+        int vectorY = targetPoint.y() - beforePoint.y();
+
+        Path path = new Path();
+
+        if (vectorX == 0) {
+            int unitVectorY = vectorY / Math.abs(vectorY);
+            for (int i = 0; i < Math.abs(vectorY); i++) {
+                path.addPoint(new Point(targetPoint.x(), targetPoint.y() - unitVectorY * i));
+            }
+        }
+
+        if (vectorY == 0) {
+            int unitVectorX = vectorX / Math.abs(vectorX);
+            for (int i = 0; i < Math.abs(vectorX); i++) {
+                path.addPoint(new Point(targetPoint.x() - unitVectorX * i, targetPoint.y()));
+            }
+        }
+
+        return path;
+    }
+
+    @Override
+    public boolean canMove(Map<Piece, Boolean> piecesOnPathWithTargetOrNot) {
+        if (piecesOnPathWithTargetOrNot.size() >= 3) {
+            return false;
+        }
+        if (piecesOnPathWithTargetOrNot.size() == 2) {
+            if (piecesOnPathWithTargetOrNot
+                    .keySet()
+                    .stream()
+                    .anyMatch(piece -> piece instanceof Pho)) {
+                return false;
+            }
+            if (piecesOnPathWithTargetOrNot.values()
+                    .stream()
+                    .noneMatch(isTargetPoint -> isTargetPoint)) {
+                return false;
+            }
+            if (piecesOnPathWithTargetOrNot.entrySet()
+                    .stream()
+                    .filter(Entry::getValue)
+                    .findFirst()
+                    .orElseThrow(() -> new RuntimeException("[ERROR] 종점에 위치한 장애물이 존재하지 않습니다.\n"))
+                    .getKey()
+                    .getTeam() == this.team) {
+                return false;
+            }
+            return true;
+        }
+        if (piecesOnPathWithTargetOrNot.size() == 1) {
+            if (piecesOnPathWithTargetOrNot.values()
+                    .stream()
+                    .findFirst()
+                    .get() ||
+                    piecesOnPathWithTargetOrNot.keySet()
+                            .stream()
+                            .findFirst()
+                            .get() instanceof Pho) {
+                return false;
+            }
+            return true;
+        }
+        return true;
```

Comment:
위와 같은 조건문은 조건을 바로 리턴하는 것과 동일합니다.
이 부분도 어떤 조건인지 변수로 표현해주면 더 좋고요.

```suggestion
         if (piecesOnPathWithTargetOrNot.size() == 1) {
            return !(piecesOnPathWithTargetOrNot.values()
                    .stream()
                    .findFirst()
                    .get() ||
                    piecesOnPathWithTargetOrNot.keySet()
                            .stream()
                            .findFirst()
                            .get() instanceof Pho);
        }
```

Reply:
피드백 반영하여 조건 별 함수화를 통해 canMove 함수 가독성을 높였습니다! 감사합니다!